File: src/main/java/com/procurement/storage/model/entity/FileEntity.java
package com.procurement.storage.model.entity;

import lombok.Getter;
import lombok.Setter;
import org.springframework.data.cassandra.core.cql.Ordering;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.Column;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;
import org.springframework.data.cassandra.core.mapping.Table;

import java.util.Date;
import java.util.UUID;

@Getter
@Setter
@Table("storage_files")
public class FileEntity {

    @PrimaryKeyColumn(name = "file_id", type = PrimaryKeyType.PARTITIONED)
    private UUID id;

    @PrimaryKeyColumn(name = "file_is_open", type = PrimaryKeyType.CLUSTERED)
    private Boolean isOpen;

    @PrimaryKeyColumn(name = "date_modified", type = PrimaryKeyType.CLUSTERED, ordering = Ordering.DESCENDING)
    private Date dateModified;

    @Column(value = "date_published")
    private Date datePublished;

    @Column(value = "file_hash")
    private String hash;

    @Column(value = "file_weight")
    private Integer weight;

    @Column(value = "file_name")
    private String fileName;

    @Column(value = "file_on_server")
    private String fileOnServer;

    @Column(value = "file_url")
    private String url;

    @Column(value = "file_desc")
    private String description;

}

File: src/main/kotlin/com/procurement/storage/model/entity/FileEntity.kt
package com.procurement.storage.model.entity

import org.springframework.data.cassandra.core.cql.PrimaryKeyType
import org.springframework.data.cassandra.core.mapping.Column
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn
import org.springframework.data.cassandra.core.mapping.Table
import java.util.*

@Table(value = "storage_files")
data class FileEntity(

        @PrimaryKeyColumn(name = "file_id", type = PrimaryKeyType.PARTITIONED)
        var id: UUID,

        @PrimaryKeyColumn(name = "file_is_open", type = PrimaryKeyType.CLUSTERED)
        var isOpen: Boolean,

        @Column(value = "date_modified")
        var dateModified: Date?,

        @Column(value = "date_published")
        var datePublished: Date?,

        @Column(value = "file_hash")
        var hash: String?,

        @Column(value = "file_weight")
        var weight: Long?,

        @Column(value = "file_name")
        var fileName: String?,

        @Column(value = "file_on_server")
        var fileOnServer: String?,

        @Column(value = "file_owner")
        var owner: String?

//    fun getDateModified(): LocalDateTime {
//        return LocalDateTime.ofInstant(dateModified!!.toInstant(), ZoneOffset.UTC)
//    }
//
//    fun setDateModified(dateModified: LocalDateTime) {
//        this.dateModified = Date.from(dateModified.toInstant(ZoneOffset.UTC))
//    }
//
//    fun getDatePublished(): LocalDateTime {
//        return LocalDateTime.ofInstant(datePublished!!.toInstant(), ZoneOffset.UTC)
//    }
//
//    fun setDatePublished(datePublished: LocalDateTime) {
//        this.datePublished = Date.from(datePublished.toInstant(ZoneOffset.UTC))
//    }
)
